pandas/core/arrays/period.py

```diff
@@ -338,6 +347,19 @@ def __array__(self, dtype=None):
         # overriding DatetimelikeArray
         return np.array(list(self), dtype=object)
 
+    def __arrow_array__(self, type=None):
+        """
+        Convert myself into a pyarrow Array.
+        """
+        import pyarrow as pa
+
+        if type is not None and not isinstance(type, PeriodType):
+            raise TypeError("not supported")
```

Comment:
Include `type` in the error message?

Reply:
What about when `type` is an instance of PeriodType? You seem to ignore it.

I don't know what to do when someone passes a different freq than `self.freq`.

Reply:
Ah, I should check that the `freq` is the same in that case (basically right now I just want to disallowing a type that is different from self, as a starting point)